File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int ex = brown + yellow;
        int s = 0;
        for (int i = 1; i < ex; i++) {
            if (ex % i == 0) {
                s = ex / i;
                if (i <= s && (s - 2) * (i - 2) == yellow) {
                    answer[0] = s;
                    answer[1] = i;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    const width = brown + yellow;
    let s = 0;
    
    for (let i = 0; i < width; i++) {
        if (width % i === 0) {
            s = width / i;
            if (i <= s && (s - 2) * (i - 2) === yellow) {
                answer[0] = s;
                answer[1] = i;
                break;
            }
        }
    }
    
    return answer;
}